lib/spack/spack/test/llnl/util/filesystem.py

```diff
@@ -1178,16 +1184,16 @@ def _check_find_links(root, locations):
     assert set(fs.find_max_depth(root, "l3-f3")) == {locations["l3-f3-link-l1"]}
 
 
-@pytest.mark.parametrize(
-    "use_junctions",
-    [
-        False,
-        pytest.param(
-            True,
-            marks=pytest.mark.skipif(sys.platform != "win32", reason="Only Windows has junctions"),
-        ),
-    ],
-)
-def test_find_max_depth_symlinks(tmpdir, use_junctions):
-    root, locations = dir_structure_with_things_to_find_links(tmpdir, use_junctions=use_junctions)
-    _check_find_links(root, locations)
+def test_find_max_depth_multiple_and_repeated_entry_points(complex_dir_structure):
+    root, locations = complex_dir_structure
+
+    fst = str(pathlib.Path(root) / "l1-d1" / "l2-d1")
+    snd = str(pathlib.Path(root) / "l1-d2")
+    nonexistent = str(pathlib.Path(root) / "nonexistent")
+
+    assert fs.find([fst, snd, fst, snd, nonexistent], ["l*-f*"], max_depth=1) == [
+        locations["l2-f1"],
+        locations["l4-f1"],
+        locations["l4-f2-full"],
+        locations["l3-f3-full"],
+    ]
```

Comment:
`find` doesn't guarantee returning results in a particular order beyond

> If ``globs`` is a list, files matching earlier entries are placed in the return value before files matching later entries.

I recommend using set-equality here.